GetandTakeAPI/Program.cs

```diff
@@ -0,0 +1,98 @@
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using AutoMapper;
+using GetandTake.Business.DependencyResolvers.Autofac;
+using GetandTake.Business.Services.AutoMapper;
+using GetandTake.Configuration.Extensions;
+using GetandTake.Configuration.Settings;
+using GetandTake.Core.DependencyResolvers;
+using GetandTake.Core.Extensions;
+using GetandTake.Core.Utilities.IoC;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerUI;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+builder.Services.AddDependencyResolvers(new ICoreModule[]{
+    new CoreModule()
+});
+
+builder.Services.AddSingleton(new MapperConfiguration(mapperConfig =>
+                                                      mapperConfig.AddProfile(new AutoMapperProfile())).CreateMapper());
+builder.Services.AddResponseCaching(options =>
+{
+    options.MaximumBodySize = 1024;
+    options.UseCaseSensitivePaths = true;
+});
+
+var databaseSettings = builder.Configuration.GetSection(nameof(AppSettings.Database)).Get<DatabaseSettings>();
+var productSettings = builder.Configuration.GetSection(nameof(AppSettings.Products)).Get<ProductsSettings>();
+var hostSettings = builder.Configuration.GetSection(nameof(AppSettings.Host)).Get<HostSettings>();
+var logFilterSettings = builder.Configuration.GetSection(nameof(AppSettings.LoggingParameters)).Get<LogFilterSettings>();
+
+var appSettings = new AppSettings
+{
+    Database = databaseSettings,
+    Products = productSettings,
+    Host = hostSettings,
+    LoggingParameters = logFilterSettings
+};
+
+builder.Services.RegisterServices(appSettings);
+builder.Services.RegisterDatabase(appSettings);
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen(swaggerGen =>
+{
+    swaggerGen.SwaggerDoc("v1", new OpenApiInfo
+    {
+        Title = "GetandTake.API",
+        Version = "v1",
+        Description = "NET 6",
+        Contact = new OpenApiContact
+        {
+            Name = "GetandTake API Project"
+        }
+    });
+});
+
+builder.Services.AddCors(policy => policy
+.AddPolicy("getandtakeapp", builder =>
```

Comment:
You can check [this](https://github.com/dotnet-fizzyy/ASP.Net-Identity/blob/88d88cce9375c505e0d83b6f1037975b7f622b20/src/IdentityWebApi/Startup/WebAppConfigurationExtensions.cs#L31) flow how to build your CORS policy that allows all available options. Usage of policies here is quite redundant because you have just a single policy, it makes sense if you have different origins configurations, different requirements for endpoints and etc

Reply:
for API project, it's done.